packages/terra-table/src/Table.jsx

```diff
@@ -0,0 +1,352 @@
+      const lastPinnedColumnIndex = pinnedColumns.length - 1;
+
+      renderedColumns.slice(0, lastPinnedColumnIndex).forEach((pinnedColumn) => {
+        cumulativeOffset += pinnedColumn.width;
+        offsetArray.push(cumulativeOffset);
+      });
+    }
+
+    setPinnedColumnOffsets(offsetArray);
+
+    const currentSortedColumn = renderedColumns.find((column) => !!column.sortIndicator);
+    if (sortedColumn.current) {
+      const previousSortedColumn = { ...sortedColumn.current };
+
+      if (previousSortedColumn.id !== currentSortedColumn.id
+      || previousSortedColumn.sortIndicator !== currentSortedColumn.sortIndicator) {
+        setCellAriaLiveMessage(intl.formatMessage(
+          { id: 'Terra.table.sort-direction-changed' },
+          {
+            column: currentSortedColumn.displayName,
+            sortDirection: currentSortedColumn.sortIndicator,
+          },
+        ));
+      }
+    }
+
+    sortedColumn.current = currentSortedColumn;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [renderedColumns]);
+
+  // -------------------------------------
+
+  // -------------------------------------
+  // event handlers
+
+  const handleColumnSelect = useCallback((columnId) => {
+    if (onColumnSelect) {
+      onColumnSelect(columnId);
+    }
+  }, [onColumnSelect]);
+
+  const onResizeMouseDown = useCallback((event, index, resizeColumnWidth) => {
+    // Store current table and column values for resize calculations
+    tableWidth.current = tableRef.current.offsetWidth;
+    activeColumnPageX.current = event.pageX;
+    activeColumnWidth.current = resizeColumnWidth;
+
+    // Set the active index to the selected column
+    setActiveIndex(index);
+  }, []);
+
+  const onMouseMove = (event) => {
+    if (activeIndex == null) {
+      return;
+    }
+
+    // Ensure the new column width falls within the range of the minimum and maximum values
+    const diffX = event.pageX - activeColumnPageX.current;
+    const { minimumWidth, maximumWidth } = renderedColumns[activeIndex];
+    const newColumnWidth = Math.min(Math.max(activeColumnWidth.current + diffX, minimumWidth), maximumWidth);
+
+    // Update the width for the column in the state variable
+    const newColumns = [...renderedColumns];
+    newColumns[activeIndex].width = newColumnWidth;
+    setRenderedColumns(newColumns);
+
+    // Update the column and table width
+    tableRef.current.style.width = `${tableWidth + (newColumnWidth - activeColumnWidth.current)}px`;
+  };
+
+  const onMouseUp = () => {
+    if (onColumnResize) {
+      onColumnResize(renderedColumns[activeIndex].id, renderedColumns[activeIndex].width);
+    }
+    // Remove active index
+    setActiveIndex(null);
+  };
+
+  // -------------------------------------
+
+  return (
+    <div
+      className={cx('table-container')}
+      ref={tableContainerRef}
+      // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
+      tabIndex={isContainerFocusable ? 0 : undefined}
```

Comment:
We should probably handle the scrolling in a separate JIRA.  We are adding too much to one PR.  However, I am also not sure this container should be focusable when in a grid context.

Reply:
Removed all logic about manually managing container focus. Although we do set the tabIndex to 0 in DataGrid so we'll probably flex it to 0 on grid context here.

Reply:
The DataGrid is setting it to 0 to control focus as it is a composite widget.  The Terra Table will not be a composite widget so we don't need to control focus in the same way.